includes/Export.php

```diff
@@ -194,42 +230,78 @@ protected function do_list_authors( $cond ) {
 		foreach ( $res as $row ) {
 			$this->author_list .= "<contributor>" .
 				"<username>" .
-				htmlentities( $row->rev_user_text )  .
+				htmlentities( $row->rev_user_text ) .
 				"</username>" .
 				"<id>" .
 				$row->rev_user .
 				"</id>" .
 				"</contributor>";
 		}
 		$this->author_list .= "</contributors>";
-		wfProfileOut( __METHOD__ );
 	}
 
+	/**
+	 * @param string $cond
+	 * @throws MWException
+	 * @throws Exception
+	 */
 	protected function dumpFrom( $cond = '' ) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this function "dumpFrom" is 21 which is greater than 20 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This function "dumpFrom" has 167 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)